subprojects/docs/src/snippets/java/toolchain-task/groovy/build.gradle

```diff
@@ -1,44 +1,44 @@
 import javax.inject.Inject;
 
+// tag::custom-toolchain-task-with-java[]
 abstract class CustomTaskUsingToolchains extends DefaultTask {
 
     @Nested
-    abstract Property<JavaLauncher> getLauncher()
+    abstract Property<JavaLauncher> getLauncher() // <1>
 
-    @Inject
     CustomTaskUsingToolchains() {
-        // Access the default toolchain
-        def toolchain = project.getExtensions().getByType(JavaPluginExtension.class).toolchain
-
-        // acquire a provider that returns the launcher for the toolchain
-        JavaToolchainService service = project.getExtensions().getByType(JavaToolchainService.class)
-        Provider<JavaLauncher> defaultLauncher = service.launcherFor(toolchain);
-
-        // use it as our default for the property
-        launcher.convention(defaultLauncher);
+        def toolchain = project.extensions.getByType(JavaPluginExtension.class).toolchain // <2>
```

Comment:
I think this shouldn't be done in a task (or should it)? Shouldn't the corresponding plugin set this up?

Reply:
I guess it is up to the plugin author how to wire thing together. I am not entirely sure what is the best practice currently.

Ultimately, I didn't change the content of this section. Only modernized some parts.

I'll take another look if we could provide a better organization for this example here. It should help that it's already split in multiple snippets when rendered in the doc.

Reply:
Having the plugin set conventions for the tasks it creates is the way we've been promoting.  For a small example I don't think it's necessary to write a full Plugin class though.